File: src/main/java/nu/peg/slack/pt/api/slack/model/InteractivePostData.java
package nu.peg.slack.pt.api.slack.model;

import com.google.gson.Gson;

import javax.ws.rs.FormParam;

import lombok.Data;
import lombok.Getter;

@Data
public class InteractivePostData {

    @FormParam("payload")
    private String payload;

    @Getter(lazy = true)
    private final InteractivePayload payloadObject = getAsObject();

    private InteractivePayload getAsObject() {
        return new Gson().fromJson(payload, InteractivePayload.class);
    }
}

File: src/main/kotlin/nu/peg/slack/pt/api/slack/model/InteractivePostData.kt
package nu.peg.slack.pt.api.slack.model

import com.google.gson.Gson

import javax.ws.rs.FormParam

import lombok.Data
import lombok.Getter

@Data
class InteractivePostData {

    @FormParam("payload")
    private val payload: String? = null

    @Getter(lazy = true)
    private val payloadObject = asObject

    private val asObject: InteractivePayload
        get() = Gson().fromJson(payload, InteractivePayload::class.java)
}
